account_chart_update/tests/test_account_chart_update.py

```diff
@@ -466,3 +466,18 @@ def test_00_matching(self):
         self.assertTrue(list(self.fp.get_external_id().values())[0])
         self.assertEqual(fp_rec.get_external_id().get(fp_id), expected_xmlid)
         wizard.unlink()
+
+    def test_01_archived_fiscal_position(self):
+        # Test wizard won't duplicate existing fiscal positions when archived
+        self.fp_template.tax_ids.tax_dest_id = self.tax_template.id
+        self.fp_template.tax_ids.tax_src_id = self.tax_template.id
+        fiscal_position = self.env["account.fiscal.position"].search(
+            [("name", "=", self.fp_template.name), ("company_id", "=", self.company.id)]
+        )
+        fiscal_position.active = False
+        wizard = self.wizard_obj.create(self.wizard_vals)
+        wizard.action_find_records()
+        wizard.action_update_records()
+        wizard.unlink()
+        self.assertTrue(fiscal_position.exists())
+        self.assertFalse(fiscal_position.active)
```

Comment:
Please add again the assertion that checks tax mappings are OK.

Reply:
The test is failing:

```
2024-11-21 08:11:35,717 926 ERROR odoo odoo.addons.account_chart_update.tests.test_account_chart_update: FAIL: TestAccountChartUpdate.test_01_archived_fiscal_position
Traceback (most recent call last):
  File "/__w/account-financial-tools/account-financial-tools/account_chart_update/tests/test_account_chart_update.py", line 483, in test_01_archived_fiscal_position
    self.assertFalse(fiscal_position.active)
AssertionError: True is not false

Reply:
Oh! I see. I just found out that as the account.fiscal.position.template doesn't have the 'active' field, so when the wizard tries to update it makes it active again as is the default value for such field.

So now the error has been fixed but I would raise this question: Is it desirable to make those fiscal fiscal positions active again each time you run the wizard?

Reply:
> So now the error has been fixed but I would raise this question: Is it desirable to make those fiscal fiscal positions active again each time you run the wizard?

No, disabled FPs should remain disabled.